src/app/logic/player.js

```diff
@@ -0,0 +1,44 @@
+export class Player {
+  constructor() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+    this.endGameData = {};
+  }
+
+  registerAnswer(correctAnswer, playerAnswer, isanswerCorrect) {
+    if (this.playerAnswer === false) {
+      throw new Error(
+        'error with playerAnswer, playerAnswer is null or undefined',
+      );
+    }
+    if (isanswerCorrect) {
+      this.correctAnswersScore += 1;
+    }
+    this.detailQuestionData.push({
+      playerAnswer,
+      isAnswerCorrect: isanswerCorrect,
+      answer: correctAnswer,
+    });
+    this.amountOfAnsweredQuestions += 1;
+  }
+
+  clearGameData() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+  }
+
+  createEndGameData() {
+    this.endGameData.correctAnswersScore = this.correctAnswersScore;
```

Comment:
I wonder if we need to keep duplicated information. Perhaps you could just return a new object with all the data you need?
`return {
  correctAnswersScore: this.correc....,
  amountOfAnsweresQ: this.amoutnOF....,
  detailQuestion: this.detail....
  }